spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -373,4 +386,371 @@ public void testNotIn() {
+    Predicate predicate = new Predicate("=", expressions(udf, literalValue));
+
+    Expression icebergExpr = SparkV2Filters.convert(predicate);
+    Assertions.assertThat(icebergExpr).isNull();
+  }
+
+  private <T> void testUDF(
+      org.apache.spark.sql.connector.expressions.Expression expression,
+      UnboundTerm<T> item,
+      T value,
+      DataType dataType) {
+    org.apache.spark.sql.connector.expressions.Expression[] attrOnly = expressions(expression);
+
+    LiteralValue literalValue = new LiteralValue(value, dataType);
+    org.apache.spark.sql.connector.expressions.Expression[] attrAndValue =
+        expressions(expression, literalValue);
+    org.apache.spark.sql.connector.expressions.Expression[] valueAndAttr =
+        expressions(literalValue, expression);
+
+    Predicate isNull = new Predicate("IS_NULL", attrOnly);
+    Expression expectedIsNull = Expressions.isNull(item);
+    Expression actualIsNull = SparkV2Filters.convert(isNull);
+    Assertions.assertThat(actualIsNull.toString()).isEqualTo(expectedIsNull.toString());
+
+    Predicate isNotNull = new Predicate("IS_NOT_NULL", attrOnly);
+    Expression expectedIsNotNull = Expressions.notNull(item);
+    Expression actualIsNotNull = SparkV2Filters.convert(isNotNull);
+    Assertions.assertThat(actualIsNotNull.toString()).isEqualTo(expectedIsNotNull.toString());
+
+    Predicate lt1 = new Predicate("<", attrAndValue);
+    Expression expectedLt1 = Expressions.lessThan(item, value);
+    Expression actualLt1 = SparkV2Filters.convert(lt1);
+    Assertions.assertThat(actualLt1.toString()).isEqualTo(expectedLt1.toString());
+
+    Predicate lt2 = new Predicate("<", valueAndAttr);
+    Expression expectedLt2 = Expressions.greaterThan(item, value);
+    Expression actualLt2 = SparkV2Filters.convert(lt2);
+    Assertions.assertThat(actualLt2.toString()).isEqualTo(expectedLt2.toString());
+
+    Predicate ltEq1 = new Predicate("<=", attrAndValue);
+    Expression expectedLtEq1 = Expressions.lessThanOrEqual(item, value);
+    Expression actualLtEq1 = SparkV2Filters.convert(ltEq1);
+    Assertions.assertThat(actualLtEq1.toString()).isEqualTo(expectedLtEq1.toString());
+
+    Predicate ltEq2 = new Predicate("<=", valueAndAttr);
+    Expression expectedLtEq2 = Expressions.greaterThanOrEqual(item, value);
+    Expression actualLtEq2 = SparkV2Filters.convert(ltEq2);
+    Assertions.assertThat(actualLtEq2.toString()).isEqualTo(expectedLtEq2.toString());
+
+    Predicate gt1 = new Predicate(">", attrAndValue);
+    Expression expectedGt1 = Expressions.greaterThan(item, value);
+    Expression actualGt1 = SparkV2Filters.convert(gt1);
+    Assertions.assertThat(actualGt1.toString()).isEqualTo(expectedGt1.toString());
+
+    Predicate gt2 = new Predicate(">", valueAndAttr);
+    Expression expectedGt2 = Expressions.lessThan(item, value);
+    Expression actualGt2 = SparkV2Filters.convert(gt2);
+    Assertions.assertThat(actualGt2.toString()).isEqualTo(expectedGt2.toString());
```

Comment:
Could you try using `ExpressionUtil.equivalent`? I think that's probably a better way to check.

Reply:
Updated.